core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/DeleteExpiredMetrics.java

```diff
@@ -0,0 +1,114 @@
+
+    private long metricExpirationDelay;
+
+    public DeleteExpiredMetrics(MetricsService metricsService, RxSession session,
+            ConfigurationService configurationService, int metricExpirationDelayInDays) {
+        this.metricsService = metricsService;
+        this.session = session;
+        this.configurationService = configurationService;
+
+        findEligibleTenants = session.getSession()
+                .prepare("SELECT DISTINCT tenant_id, type FROM metrics_expiration_idx");
+        findEligibleMetrics = session.getSession()
+                .prepare(
+                        "SELECT tenant_id, type, metric, time FROM metrics_expiration_idx WHERE tenant_id = ? AND type = ?");
+        findUnexpiredDataPoints = session.getSession()
+                .prepare(
+                        "SELECT * FROM data WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = 0 LIMIT 1;");
+
+        this.metricExpirationDelay = metricExpirationDelayInDays * 24 * 3600 * 1000L;
+    }
+
+    @Override
+    public Completable call(JobDetails jobDetails) {
+        String unparsedConfigExpirationTime = jobDetails.getParameters().get("expirationTimestamp");
+        Long configuredExpirationTime = null;
+        if (unparsedConfigExpirationTime != null && !unparsedConfigExpirationTime.isEmpty()) {
+            try {
+                configuredExpirationTime = Long.parseLong(unparsedConfigExpirationTime);
+            } catch (Exception exp) {
+                //do nothing just use the default configuration
+            }
+        }
+
+        long expirationTime = (configuredExpirationTime != null ? configuredExpirationTime
+                : DateTimeService.now.get().getMillis()) - metricExpirationDelay;
+
+        Observable<MetricId<?>> expirationIndexResults = session.execute(findEligibleTenants.bind())
+                .flatMap(Observable::from)
+                .flatMap(row -> session.execute(findEligibleMetrics.bind(row.getString(0), row.getByte(1))))
+                .flatMap(Observable::from)
+                .filter(row -> row.getTimestamp(3).getTime() < expirationTime)
+                .map(row -> new MetricId<>(row.getString(0), MetricType.fromCode(row.getByte(1)), row.getString(2)));
+
+        //If the compression job is disabled then check the data point table for data
+        String compressJobEnabledConfig = configurationService.load(CompressData.CONFIG_ID, "enabled").toBlocking()
+                .firstOrDefault(null);
+
+        boolean compressJobEnabled = false;
+        if (compressJobEnabledConfig != null && !compressJobEnabledConfig.isEmpty()) {
+            try {
+                compressJobEnabled = Boolean.parseBoolean(compressJobEnabledConfig);
+            } catch (Exception e) {
+                //do nothing, assume the compression job is disabled
+            }
+        }
+        if (!compressJobEnabled) {
+            expirationIndexResults = expirationIndexResults
+                    .flatMap(r -> session
```

Comment:
If there is no entry in the expiration index then that is good, it means the data will automatically expire based on the TTL. We are only concerned about manually cleaning that does not have a TTL (in general that is indexes). For that case, an entry in the expiration index is created every a metric is intentionally created or when tags are inserted or updated. 

All what the compress job do is extend the expiration time. If the compress job is disabled, then we have no way of reliably extending the expiration time, hence we need to query and check if there is unexpired data.